src/LinAlg/hiopLinSolverCholCuSparse.cpp

```diff
@@ -161,9 +161,23 @@ bool hiopLinSolverCholCuSparse::do_symb_analysis(const size_type n,
     const int64_t *options = nullptr; //use default METIS options
     ret = cusolverSpXcsrmetisndHost(h_cusolver_, n, nnz, mat_descr_, rowptr, colind, options, perm);
     assert(ret == CUSOLVER_STATUS_SUCCESS);
-  } else if ("symamd" == ordering) {
+  } else if("symamd-cuda" == ordering) {
     ret = cusolverSpXcsrsymamdHost(h_cusolver_, n, nnz, mat_descr_, rowptr, colind, perm);
     assert(ret == CUSOLVER_STATUS_SUCCESS);
+  } else if("symamd-eigen" == ordering) {
+
+    Eigen::Map<SparseMatrixCSR> M(n,
+                                  n,
+                                  nnz,
+                                  const_cast<int*>(rowptr),
+                                  const_cast<int*>(colind),
+                                  const_cast<double*>(value));
+
+    PermutationMatrix P;
+    Ordering ordering;
+    ordering(M.selfadjointView<Eigen::Upper>(), P);
+    memcpy(perm, P.indices().data(), n*sizeof(int));
+
```

Comment:
Do we need to have `#ifdef HIOP_USE_EIGEN` in this section? Can `Eigen` be undefined?

Reply:
indeed, it can be